services/debug_tools.py

```
@@ -0,0 +1,35 @@
+
+    async def export_chat_history(self, admin_token: str, chat_id: str) -> str:
+        """Export recent messages for any Telegram chat when the shared token matches."""
+        if admin_token != ADMIN_DEBUG_TOKEN:
+            return "Invalid admin token."
+
+        rows = await MessageRepository.get_messages_for_chat_id_raw(chat_id)
+        return json.dumps(rows, default=str)
+
+    def fetch_url_preview(self, target_url: str) -> str:
+        """Fetch a remote URL and return a small preview of the response body."""
+        with urlopen(target_url, timeout=5) as response:
```

Ruff (0.14.14)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)